### PR TITLE
Expand navigator when select an element inside container

### LINKED ISSUE #156: Expand navigator when select an element inside container
Steps:
- wrap couple of elements in container
- collapse container in navigator
- select an element on canvas

Actual:
- nothing selected at Navigator

Expected:
- the container is expanded, the element is selected

### PR DESCRIPTION
**Pull request by drucelweisse**
Closes #156 

Screencast:

https://user-images.githubusercontent.com/36012972/209449708-a590761e-eb8c-49a3-95e8-415b9ddc13c9.mov

### CODE REVIEW

File: VoiceOver Designer/Features/Sources/Navigator/NavigatorController.swift [hunk 1]
```diff
@@ -55,10 +55,33 @@ public class NavigatorController: NSViewController {
         
         updateAttributedLabel(for: model, isSelected: true)
         
+        
         outlineView.selectRowIndexes(IndexSet(integer: index),
```
Comment (akaDuality) [inline VoiceOver Designer/Features/Sources/Navigator/NavigatorController.swift:59]: When we select container here is `index == -1`. Probably we can check that and separate two flow?

File: VoiceOver Designer/Features/Sources/Navigator/NavigatorController.swift [hunk 2]
```diff
@@ -55,10 +55,33 @@ public class NavigatorController: NSViewController {
...
+        expandAndSelect(model)
+        
+        
     }
     
+    private func expandAndSelect(_ element: any AccessibilityView) {
+        guard case let .element(description) = element.cast else { return }
+        guard let container = document.controls.extractContainers()
+            .first(where: { $0.contains(description) }),
+        let childIndex = container.elements.firstIndex(of: description) else { return }
```
Comment (akaDuality) [inline VoiceOver Designer/Features/Sources/Navigator/NavigatorController.swift:71]: Looks like we can move this logic to `Document`

Comment (akaDuality) [reply VoiceOver Designer/Features/Sources/Navigator/NavigatorController.swift:71]: Or ever here is a function that allows to find parent inside OutlineView: `outlineView.parent(forItem: description)`

**Review by akaDuality — CHANGES_REQUESTED** [review]

File: VoiceOver Designer/Features/Sources/Navigator/NavigatorController.swift (hunk 2 above)
Comment (drucelweisse) [reply VoiceOver Designer/Features/Sources/Navigator/NavigatorController.swift:71]: It seems that OutlineView doesn't know about child parent when it collapsed, as function mentioned above always return nil. Probably we should rely on datasource. I'll move that in document logic into Document

Comment (akaDuality) [reply VoiceOver Designer/Features/Sources/Navigator/NavigatorController.swift:71]: I've checked it and it works. You should pass description, not model

Comment (akaDuality) [reply VoiceOver Designer/Features/Sources/Navigator/NavigatorController.swift:71]: Screenshot 2022-12-25 at 12 59 03

Comment (drucelweisse) [reply VoiceOver Designer/Features/Sources/Navigator/NavigatorController.swift:71]: That's strange I always get nil with that setup

```swift
guard case let .element(description) = element.cast else { return }
let container = outlineView.parent(forItem: description)
// Expanding Container, breakpoint here with po container
outlineView.expandItem(container, expandChildren: false)
        
let rowToSelect = outlineView.row(forItem: element)
guard isValid(row: rowToSelect) else { return }
outlineView.selectRowIndexes(IndexSet(integer: rowToSelect),
                                     byExtendingSelection: false)
```

https://user-images.githubusercontent.com/36012972/209461012-d9843b81-4813-4297-8bda-8118f9407dee.mov

Comment (akaDuality) [reply VoiceOver Designer/Features/Sources/Navigator/NavigatorController.swift:71]: I got it, it because parent was expanded at the line of my sample

Comment (drucelweisse) [reply VoiceOver Designer/Features/Sources/Navigator/NavigatorController.swift:71]: oh sorry didn't see second comment)

Comment (drucelweisse) [reply VoiceOver Designer/Features/Sources/Navigator/NavigatorController.swift:71]: Yeah, that's correct but if we try to find it without model and rely only on OutlineView, parent can't be found

**Review by akaDuality — APPROVED** [review]

**Pull request merged**